zookeeper-server/src/main/java/org/apache/zookeeper/server/util/DigestCalculator.java

```diff
@@ -37,19 +37,28 @@
 
     // The hardcoded digest version, should bump up this version whenever
     // we changed the digest method or fields.
-    //
-    // Defined it as Integer to make it able to be changed in test via reflection
-    public static final Integer DIGEST_VERSION = 2;
+    private static final int DIGEST_VERSION = 2;
 
+    public static final DigestCalculator DIGEST_CALCULATOR;
```

Comment:
What if we make `DigestCalculator` interface, remove final from here and mock it easily in tests?
What's the benefit of making it final?

Reply:
I think that would work too, and mocking is more elegant than using plain reflection in the tests. (I think with Mockito you can even mock the class and you don't even need an intreface)

The only issue with the removal of the `final` modifier is that from that point anywhere from the production code you can easily change this singleton. But we can write a comment that it is to be changed only from tests...